src/main/java/net/smacke/jaydio/OpenFlags.java

```diff
@@ -15,20 +15,38 @@
  */
 package net.smacke.jaydio;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+
 /**
  * Constants for {@link DirectIoLib#oDirectOpen(String, boolean)}. </p>
  *
  * @author smacke
  *
  */
 public final class OpenFlags {
-    public static final int O_RDONLY = 00;
-    public static final int O_WRONLY = 01;
-    public static final int O_RDWR = 02;
-    public static final int O_CREAT = 0100;
-    public static final int O_TRUNC = 01000;
-    public static final int O_DIRECT = 040000;
-    public static final int O_SYNC = 04000000;
+    public static final int O_RDONLY = getOpenFlags("O_RDONLY");
+    public static final int O_WRONLY = getOpenFlags("O_WRONLY");
+    public static final int O_RDWR = getOpenFlags("O_RDWR");
+    public static final int O_CREAT = getOpenFlags("O_CREAT");
+    public static final int O_TRUNC = getOpenFlags("O_TRUNC");
+    public static final int O_DIRECT = getOpenFlags("O_DIRECT");
+    public static final int O_SYNC = getOpenFlags("O_SYNC");
 
     private OpenFlags() {}
+
+    private static int getOpenFlags(String s) {
```

Comment:
Please provide default values for all the supported flag types and return them in case dynamic invocation fails.

Reply:
Some macro values may be fixed across OSes. But @fcyu found that the `O_DIRECT` macro was different for 2 OSes, causing an error, that is why we query the macro value with the compiler.

Reply:
@guoci I'm not saying that querying is not needed/bad. I'm saying that the way you guys implemented it will break the library for the majority of use-cases. I'm suggesting that instead of just crashing the library when there's no `/bin/sh` or no `gcc` or no `libc` sources installed - fall back and return the original default values. It's a simple change.

Reply:
Added 11474fb918cb6b6d2df824837fd95bdf9c763def

Reply:
https://github.com/chhh/jaydio/commit/f2ec2e66979852384659ddb7f36326f78ffcded1 should catch all exceptions.